flake.nix

```diff
@@ -0,0 +1,54 @@
+{
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/25.05";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+  outputs = { self, nixpkgs, flake-utils }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        pkgs = nixpkgs.legacyPackages.${system};
+        ghcWithPkgs = pkgs.haskellPackages.ghcWithPackages (hpkgs: with hpkgs; [
+          colour
+          directory
+          extra
+          JuicyPixels
+          lucid2
+          optparse-generic
+          pretty-simple
+          shake
+          text
+          unix
+        ]);
+      in
+      {
+        devShells.default = pkgs.mkShell {
+          buildInputs = with pkgs; [
+            ghcWithPkgs
+            fourmolu
+            haskell-language-server
+          ];
+        };
+      } // {
+        packages = pkgs.lib.listToAttrs (map
+          (name: {
+            inherit name;
+            value =
+              let module = pkgs.lib.strings.toUpper (pkgs.lib.substring 0 1 name) + pkgs.lib.substring 1 (-1) name;
```

Comment:
This mapping is good enough for the current two outputs because they have one word names, but in general we need to replace upper case letters after the start with a dash as well as the lower case version. Unless we just accept capitals in the binary and package names.